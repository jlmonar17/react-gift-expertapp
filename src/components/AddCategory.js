import { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
    const [inputValue, setInputValue] = useState("");

    const handleChange = (e) => {
        setInputValue(e.target.value);
    };

    const handleSubmit = (e) => {
        e.preventDefault();

        if (inputValue.length > 2) {
            setCategories((category) => [inputValue, ...category]);
            setInputValue("");
        }
    };

    return (
        <form onSubmit={handleSubmit}>
            <input type="text" onChange={handleChange} value={inputValue} />
        </form>
    );
};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
};
