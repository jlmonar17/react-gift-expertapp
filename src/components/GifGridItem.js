export const GifGridItem = ({ id, title, url }) => {
    return (
        <div className="card">
            <img src={url} alt={title} key={id} />
            <p>{title}</p>
        </div>
    );
};
