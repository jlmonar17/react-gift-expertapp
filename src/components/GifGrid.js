import { GifGridItem } from "./GifGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from "prop-types";

export const GifGrid = ({ category }) => {
    const { data: images, loading } = useFetchGifs(category);

    return (
        <>
            <h3 className="animate__animated animate__pulse">{category}</h3>

            {loading && (
                <p className="animate__animated animate__flash">Cargando...</p>
            )}

            <div className="card-grid">
                {images.map((image) => {
                    return <GifGridItem key={image.id} {...image} />;
                })}
            </div>
        </>
    );
};

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
};
