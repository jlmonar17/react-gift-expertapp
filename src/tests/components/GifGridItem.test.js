import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe("Pruebas para <GifGridItem />", () => {
    const title = "Imagen categoria";
    const url = "http://localhost/image.jpg";
    const wrapper = shallow(<GifGridItem title={title} url={url} />);

    test("Prueba de que  GifGridItem se renderiza como queremos", () => {
        expect(wrapper).toMatchSnapshot();
    });

    test("Prueba que el pÃ¡rrafo tenga un texto", () => {
        const p = wrapper.find("p");

        expect(p.text()).toBe(title);
    });

    test("Prueba que la imagen tenga en src y alt, los props", () => {
        const img = wrapper.find("img");

        expect(img.prop("src")).toBe(url);
        expect(img.prop("alt")).toBe(title);
    });

    test("Prueba que el div tenga la clase animate__fadeIn", () => {
        const div = wrapper.find("div");

        // expect(div.prop("className").includes("animate__fadeIn")).toBe(true);
        expect(div.hasClass("animate__fadeIn")).toBe(true);
    });
});
