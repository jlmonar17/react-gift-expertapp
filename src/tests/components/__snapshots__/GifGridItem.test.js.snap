// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para <GifGridItem /> Prueba de que  GifGridItem se renderiza como queremos 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Imagen categoria"
    src="http://localhost/image.jpg"
  />
  <p>
    Imagen categoria
  </p>
</div>
`;
